src/Common/tests/Collections/IEnumerableTest.cs

```diff
@@ -0,0 +1,428 @@
+                                   enumerator.Reset();
+                               }
+                               else
+                               {
+                                   VerifyEnumerator(items, enumerator);
+                               }
+                           }, 5);
+            }
+        }
+
+        [Fact]
+        public void ModifyCollectionWithNewEnumerator()
+        {
+            var enumerable = GetEnumerable(GenerateItems(16));
+            var enumerator = enumerable.GetEnumerator();
+            InvalidateEnumerator(enumerable);
+            VerifyModifiedEnumerator(enumerator, null, true, false);
+        }
+
+        [Fact]
+        public void EnumerateFirstItemThenModify()
+        {
+            var items = GenerateItems(16);
+            var enumerable = GetEnumerable(items);
+            var enumerator = enumerable.GetEnumerator();
+            VerifyEnumerator(items, enumerator, 0, 1, false, true);
+            var currentItem = enumerator.Current;
+            InvalidateEnumerator(enumerable);
+            VerifyModifiedEnumerator(
+                enumerator,
+                currentItem,
+                false,
+                false);
+        }
+
+        [Fact]
+        public void EnumeratePartOfCollectionThenModify()
+        {
+            var items = GenerateItems(16);
+            var enumerable = GetEnumerable(items);
+            var enumerator = enumerable.GetEnumerator();
+            VerifyEnumerator(items, enumerator, 0, items.Length / 2, false, true);
+
+            var currentItem = enumerator.Current;
+            InvalidateEnumerator(enumerable);
+            VerifyModifiedEnumerator(enumerator, currentItem, false, false);
+        }
+
+        [Fact]
+        public void EnumerateEntireCollectionThenModify()
+        {
+            var items = GenerateItems(16);
+            var enumerable = GetEnumerable(items);
+            var enumerator = enumerable.GetEnumerator();
+            VerifyEnumerator(items, enumerator, 0, items.Length, false, true);
+
+            var currentItem = enumerator.Current;
+            InvalidateEnumerator(enumerable);
+            VerifyModifiedEnumerator(enumerator, currentItem, false, true);
+        }
+
+        [Fact]
+        public void EnumerateThenModifyThrows()
+        {
+            var items = GenerateItems(16);
+            var enumerable = GetEnumerable(items);
+            var enumerator = enumerable.GetEnumerator();
+            VerifyEnumerator(items, enumerator, 0, items.Length / 2, false, true);
+
+            var currentItem = enumerator.Current;
+            InvalidateEnumerator(enumerable);
+            Assert.Equal(currentItem, enumerator.Current);
+            Assert.Throws<InvalidOperationException>(
+                () => enumerator.MoveNext());
+            Assert.Throws<InvalidOperationException>(
+                () => enumerator.Reset());
+        }
+
+        [Fact]
+        [ActiveIssue(1170)]
```

Comment:
Is there a way to make this [ActiveIssue] only apply to the List tests?  Making this method virtual and overriding it for a List derived type might let you paint the attribute there (as long as Fact is marked as inherited); but I don't know what xUnit will do with that :)